### PR TITLE
feat: add option to enable firefox sync

### PR DESCRIPTION
**Pull request by surfaceflinger**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

ff sync is hard disabled

Issue Number: N/A

## What is the new behavior?

u can toggle it on

## Does this introduce a breaking change?

- [ ] Yes
- [x] no, it's off by default

## Other information

not sure if this enables just sync or enables 9999999999999 backdoors but its fine for me

### CODE REVIEW

File: modules/home-manager/options/misc.nix [hunk 1]
```diff
@@ -48,5 +48,10 @@ in {
         with the canvas (practically anything at this point)
       '';
     };
+
+    firefoxSync = mkEnableOption ''
+      Enable Firefox Sync. Disabled by default to prevent calling home.
```
Comment (NotAShelf) [inline modules/home-manager/options/misc.nix:53]: `mkEnableOption` [implies "Whether to enable..."](https://github.com/hsjobeki/nixpkgs/blob/migrate-doc-comments/lib/options.nix#L142) in resulting documentation. This should be just `Firefox Sync."

File: modules/home-manager/options/misc.nix [hunk 2]
```diff
@@ -48,5 +48,10 @@ in {
...
+      Quite sane to want it re-enabled as it's encrypted and convenient.
```
Comment (NotAShelf) [inline modules/home-manager/options/misc.nix:54]: Remove this line. If you'd like to mention it's encrypted, leave a source to that.

**Review by NotAShelf — CHANGES_REQUESTED** [review]
LGTM. Very minor nits regarding mkEnableOption behaviour and the new option documentation. Should be good otherwise.

File: modules/home-manager/options/misc.nix (hunk 2 above)
Comment (surfaceflinger) [reply modules/home-manager/options/misc.nix:54]: so would you prefer me to just leave "Firefox Sync" in the desc?

Comment (NotAShelf) [reply modules/home-manager/options/misc.nix:54]: yes, that would work.

**Comment by sioodmy** [comment]
you should try [xbrowsersync](https://www.xbrowsersync.org/) instead

**Comment by surfaceflinger** [comment]
I'm aware of xbrowsersync, but if Firefox Sync is encrypted and we have disabled telemetry etc. then I don't really care and I'd rather use the builtin solution

**Review by NotAShelf — APPROVED** [review]

**Comment by NotAShelf** [comment]
merging once the CI passees

**Pull request merged**